x-pack/plugins/global_search/public/services/utils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid';
+import { ApplicationStart } from 'src/core/public';
+import { GlobalSearchResult } from '../../common/types';
+import { NavigableGlobalSearchResult } from './types';
+
+const defaultPrefStorageKey = 'globalSearch:defaultPref';
+
+/**
+ * Add the `navigate` method to a {@link GlobalSearchResult | result}.
+ */
+export const addNavigate = (
+  result: GlobalSearchResult,
+  navigateToUrl: ApplicationStart['navigateToUrl']
+): NavigableGlobalSearchResult => {
+  return {
+    ...result,
+    navigate: () => navigateToUrl(result.url),
```

Comment:
if we provide url, why we need to provide `navigate`? yeah, `prefer one way to do things`

Reply:
TBH the RFC was merged before I added the `navigateToUrl` API to core, so the function was required to handle the SPA/redirect navigation logic. Now that this API exists, I have to agree that the `navigate()` function on the results doesn't add much value and add avoidable divergence between client and server 'result' types.

What should prevail, the RFC spec by keeping the method or common sense by removing it?

Reply:
I'd rather remove it to unify client & server interfaces. Let's get someone else opinion on the topic. @joshdover

Reply:
Yeah let's just remove it and update the RFC in this PR to reflect the change. Commit history serves us well here by tying the spec change to the code change 😄

Reply:
done in https://github.com/elastic/kibana/pull/66293/commits/1bb244cd10de25355826c9cfb3e852037cb0ecdd